bgpd/bgp_rpki.c

```diff
@@ -1307,6 +1321,15 @@ static int rpki_create_socket(void *_cache)
 	if (!CHECK_FLAG(vrf->status, VRF_ACTIVE) || vrf->vrf_id == VRF_UNKNOWN)
 		return -1;
 
+	/* Create FRR pthread context to allow logging */
```

Comment:
This change is not good from my perspective 

a) Every new non FRR controlled/started pthread has to know a list of things to do with this change
b) As new items are added in frr_pthread.c, someone will have to remember to go into bgp_rpki.c and add new shit. 

What I would do is create abstract the bits and bobs in frr_pthread.c into a generic startup function that is static and then have a extern function that is named something like `frr_pthread_non_controlled_startup()` that bgp_rpki.c will call.  This way bgp_rpki.c doesn't care about the internals of frr pthreads and when a new pthread is introduced by another library in the system, it will be an easy one function call to make.

Reply:
Thank you for the feedback
Changes done